### PR TITLE
Idiomatic type check

### PR DESCRIPTION
**Pull request by Avasam**
This is the third PR in a series concerning code cleanup. With the final goal to make basic type-checking validation of public methods possible, easing the addition of 3.7+ type annotations.

- Compare `None` using `is` rather than `==`
- Do static type comparisons using `isinstance`, which is the idiomatic way of type-checking in Python. Type-checkers also support type inference this way.

### CODE REVIEW

File: Pythonwin/pywin/mfc/dialog.py [hunk 1]
```diff
@@ -33,10 +33,10 @@ def __init__(self, id, dllid=None):
         dllid may be None, a dll object, or a string with a dll name"""
         # must take a reference to the DLL until InitDialog.
         self.dll = dllFromDll(dllid)
-        if type(id) == type([]):  # a template
```
Comment (mhammond) [inline Pythonwin/pywin/mfc/dialog.py:36]: isn't this (and the similar change below) wrong?

**Review by mhammond — CHANGES_REQUESTED** [review]
It's going to take some time to go through this manually - and the `import` a plus `is None` changes make it more difficult. We also need to be careful because `isinstance` and checking `type() == ...` have different semantics in the case of subclasses, which will not usually be a problem, but might sometimes.

**Comment by Avasam** [comment]
The import changes should be split off in #1986 I didn't realize this PR still contained them.
I'd definitely prefer undoing the import changes from this PR before you have to review it.

File: Pythonwin/pywin/mfc/dialog.py (hunk 1 above)
Comment (Avasam) [reply Pythonwin/pywin/mfc/dialog.py:36]: I fixed up the PR and removed the changes that shouldn't have been in there. Ready to review :)

**Comment by Avasam** [comment]
@mhammond Even if they both fall under "idiomatic type checks", would you prefer if I split off `None` checks?

**Review by mhammond — APPROVED** [review]
This looks great, thanks.

**Pull request merged**